lnwallet/chanfunding/wallet_assembler.go

```diff
@@ -260,23 +260,70 @@ func (w *WalletAssembler) ProvisionChannel(r *Request) (Intent, error) {
 		log.Infof("Performing funding tx coin selection using %v "+
 			"sat/kw as fee rate", int64(r.FeeRate))
 
+		var (
+			// allCoins referres to the entirety of coins in our
+			// wallet that are available for funding a channel.
+			allCoins []Coin
+			// manuallySelectedCoins referres to the client-side
+			// selected coins that should be considered available
+			// for funding a channel.
+			manuallySelectedCoins []Coin
+			err                   error
+		)
+
+		// outpointsToCoins maps outpoints to coins in our wallet iff
+		// these coins are existent and returns an error otherwise.
+		outpointsToCoins := func(outpoints []*wire.OutPoint) ([]Coin,
```

Comment:
I think we can lift this up to a top-level private function to cut down on the vertical distance a bit here. AFAICT, the only closure variable being used is the set of outpoints from the initial request.